Add Tooltip component (#547)
primer/brand: apps/docs/content/components/Tooltip.mdx

import {PropTableValues} from '../../src/components'

```js
import {Tooltip} from '@primer/react-brand'
```

## Examples

### Default

```jsx live
<Tooltip text="Sample tooltip content">
    <Button>Action</Button>
</Tooltip>
```

### Custom direction

By default, tooltips appear below the element. You can change the direction by passing the `direction` prop with a value of `n`, `e`, `s`, or `w`.

As long as there is enough room in the viewport, the tooltip will appear in the specified direction. If there is not enough room, the tooltip will appear in whatever direction it can place the tooltip without overflowing the viewport.

```jsx live
<Tooltip text="Sample tooltip content" direction="e">
    <Button>Action</Button>
</Tooltip>
```

### Substitute a visible label

If a ubiquitously understood icon is used in place of a text label, you can use the `type` to provide a text label for assistive technologies such as screen readers.

Use this with caution. It is generally better to use a visible label for better usability and accessibility.

```jsx live
<Tooltip text="Go fullscreen" type="label">
    <Button hasArrow={false}><ScreenFullIcon /></Button>
</Tooltip>

> ```

Reviewer: i think we should go without the live previews here, as they don't work in Gatsby

![Screenshot 2024-05-02 at 16 14 03](https://github.com/primer/brand/assets/13340707/8b27ec93-fd04-4ac5-9ff0-096cbd048338)

## Component props

| name         | type                                                              | default         | required | description                             |
| ------------ | ----------------------------------------------------------------- | --------------- | -------- | --------------------------------------- |

> | `direction`  | <PropTableValues values={['n', 'e', 's', 'w']} />                 | `'s'`           | `false`  | The text to be displayed in the tooltip |
> | `text`       | `string`                                                          | `undefined`     | `true`   | Determines whether the ActionMenu is open by default |

Reviewer: Think these are the wrong way around, and one is referencing a different component.

```suggestion
| `direction`  | <PropTableValues values={['n', 'e', 's', 'w']} />                 | `'s'`           | `false`  | The side the tooltip caret appears on |
| `text`       | `string`                                                          | `undefined`     | `true`   |  The text to be displayed in the tooltip |
```
